src/lib/vm.rs

```diff
@@ -82,10 +92,43 @@ fn binop(binop: BinOp, v1: Value, v2: Value) -> Result<Value, Interruption> {
             */
             _ => todo!(),
         },
+        WAdd => match (cont_prim_type, v1, v2) {
+            (None, _, _) => Err(Interruption::AmbiguousOperation),
+            (Some(t), Value::Nat(n1), Value::Nat(n2)) => match t {
+                PrimType::Nat => Ok(Value::Nat(n1 + n2)),
+                PrimType::Nat8 => Ok(Value::Nat(
```

Comment:
Our `Value` enum does not have `Nat8` since we couldn't determine those from parsing, but now maybe it makes sense to inject into that, rather than `Nat` here.